cosmos/config.py

```diff
@@ -0,0 +1,203 @@
+        if self.path_to_profiles_yml:
+            logger.info("Using user-supplied profiles.yml at %s", self.path_to_profiles_yml)
+            yield Path(self.path_to_profiles_yml), {}
+
+        elif self.profile_mapping:
+            profile_contents = self.profile_mapping.get_profile_file_contents(
+                profile_name=self.profile_name, target_name=self.target_name
+            )
+
+            if desired_profile_path:
+                logger.info(
+                    "Writing profile to %s with the following contents:\n%s",
+                    desired_profile_path,
+                    profile_contents,
+                )
+                # write profile_contents to desired_profile_path using yaml library
+                desired_profile_path.write_text(profile_contents)
+                yield desired_profile_path, self.profile_mapping.env_vars
+            else:
+                with tempfile.TemporaryDirectory() as temp_dir:
+                    temp_file = Path(temp_dir) / "profiles.yml"
+                    logger.info(
+                        "Creating temporary profiles.yml at %s with the following contents:\n%s",
+                        temp_file,
+                        profile_contents,
+                    )
+                    temp_file.write_text(profile_contents)
+                    yield temp_file, self.profile_mapping.env_vars
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    Contains configuration about how to execute dbt.
+
+    :param execution_mode: The execution mode for dbt. Defaults to local
+    :param dbt_executable_path: The path to the dbt executable. Defaults to dbt-ol or dbt if
+    available on the path.
+    :param dbt_cli_flags: A list of extra dbt cli flags to pass to dbt. Defaults to []
+    :param append_env: If True, append the env dictionary to the existing environment. If False,
+    replace the existing environment with the env dictionary. Defaults to False
+    :param cancel_query_on_kill: If True, cancel the query when the dbt process is killed. If False,
+    do not cancel the query when the dbt process is killed. Defaults to True
+    :param install_deps: If True, install dbt dependencies before running dbt. Defaults to False
+    :param skip_exit_code: If the dbt process exits with this exit code, do not raise an exception.
+    Defaults to None
+    """
+
+    execution_mode: ExecutionMode = ExecutionMode.LOCAL
+    dbt_executable_path: str | Path = shutil.which("dbt-ol") or shutil.which("dbt") or "dbt"
+    dbt_cli_flags: list[str] = field(default_factory=list)
+    append_env: bool = False
+    cancel_query_on_kill: bool = True
+    install_deps: bool = False
+    skip_exit_code: int | None = None
+
+
+@dataclass
+class CosmosConfig:
+    """
+    Contains all Cosmos config.
+    """
+
+    project_config: ProjectConfig
+    profile_config: ProfileConfig
+    render_config: RenderConfig = field(default_factory=RenderConfig)
```

Comment:
`render_config` is a very confusing name because it looks like a function name that renders a configuration.

Reply:
how about `renderer_config`? I don't think it's that confusing given it's one of the 4 configs